multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -185,41 +185,43 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	r.leaderNamespace = remoteCluster.GetNamespace()
 	r.leaderClusterID = string(remoteCluster.GetClusterID())
 
-	svc := &corev1.Service{}
-	if err := r.Client.Get(ctx, req.NamespacedName, &svcExport); err != nil {
-		klog.V(2).ErrorS(err, "Unable to fetch ServiceExport", "serviceexport", req.String())
-		if !apierrors.IsNotFound(err) {
-			return ctrl.Result{}, err
-		}
+	cleanup := func() error {
 		if svcInstalled {
 			err = r.handleServiceDeleteEvent(ctx, req, remoteCluster)
 			if err != nil {
-				return ctrl.Result{}, err
+				return err
 			}
 			r.installedSvcs.Delete(svcObj)
 		}
-
 		if epInstalled {
 			err = r.handleEndpointDeleteEvent(ctx, req, remoteCluster)
 			if err != nil {
-				return ctrl.Result{}, err
+				return err
 			}
 			r.installedEps.Delete(epObj)
 		}
+		return nil
+	}
+
+	if err := r.Client.Get(ctx, req.NamespacedName, &svcExport); err != nil {
+		klog.ErrorS(err, "Unable to fetch ServiceExport", "serviceexport", req.String())
```

Comment:
Ok, if it is logged in the caller, I am fine not to log here (and we should not log not found as an error if it is not).

Reply:
move it after `!apierrors.IsNotFound(err)`